insert_apis_to_db.py

```diff
@@ -0,0 +1,103 @@
+        """connect to PostgreSQL DBMS"""
+        self.dbname = dbname
+        self.user = user
+        self.password = password
+        self.connect_text = "dbname ='%s' user= '%s' password='%s'" % (self.dbname, self.user, self.password)
+        self.conn = psycopg2.connect(self.connect_text)
+        self.cursor = self.conn.cursor()
+        
+    def insertNewApiToTbl(self,records):
+        """ insert a new source information into the apilinks table """
+        createTable = """CREATE TABLE IF NOT EXISTS public.apilinks(\
+                                unique_id uuid NOT NULL DEFAULT uuid_generate_v4(),\
+                                source_name character varying(255) COLLATE pg_catalog."default",\
+                                source_apis character varying(65535) COLLATE pg_catalog."default",\
+                                access_date timestamp without time zone DEFAULT CURRENT_TIMESTAMP,\
+                                url character varying(67777) COLLATE pg_catalog."default",\
+                                api_modified_date date,\
+                                region_name character varying(255) COLLATE pg_catalog."default",\
+                                geometry geometry,\
+                                geomid character varying COLLATE pg_catalog."default" DEFAULT 'nz_1'::character varying,\
+                                api_key character varying(255) COLLATE pg_catalog."default",\
+                                query_dictionary jsonb,\
+                                geometry_col_name character varying(255) COLLATE pg_catalog."default",\
+                                UNIQUE (source_name))"""
+        insertQuery = """INSERT INTO apilinks (source_name,source_apis, url,region_name,query_dictionary,api_modified_date,geometry_col_name) VALUES (%s,%s,%s,%s,%s,%s,%s);\
+        UPDATE apilinks SET geometry =(SELECT geom FROM nz_polygons WHERE apilinks.geomid = nz_polygons.geomid)""" #adding nz polygon from another table where nz polygon is stored
+        record_to_insert = (source_name,source_apis, url, region_name,query_dictionary,api_modified_date,geometry_col_name)
+        try:
+            self.cursor.execute(createTable)
+            self.cursor.execute(insertQuery, record_to_insert)
+            self.conn.commit()
+            count = self.cursor.rowcount
+            print(count, "Record inserted successfully into datalinks table")
+            self.cursor.close()
+        except (Exception, psycopg2.DatabaseError) as error:
+            print(error)
+
+    def extractApiQueryParams(self,api):
+        """to get the key value pair of api query"""
+        parsed_api = urlparse(api)
+        params = parse_qs(parsed_api.query)
+        return json.dumps(params), api[:api.rfind(';')]+';'   
+            
+    def url_validator(self, url):
+        """check if the url entered by the user is valid"""
+        valid=validators.url(url)
+        return url if valid==True else print("Invalid url")
```

Comment:
Having this function return a `bool` would allow you to use it like this

```python
if url_invalid(url) or api_invalid(api_url):
    # prompt user for new url or raise exception
`